src/controller.py

```diff
@@ -0,0 +1,75 @@
+import os
+import subprocess
+import argparse
+import sys
+import re
+
+
+def extract_args():
+    """ Extract and return the required arguments using argparse. """
+    parser = argparse.ArgumentParser(description='Generate Living Documentation from GitHub repositories.')
+
+    parser.add_argument('--github-token', required=True, help='GitHub token for authentication.')
+    parser.add_argument('--project-state-mining', required=True, help='Enable or disable mining of project state data.')
+    parser.add_argument('--projects-title-filter', required=True, help='Filter projects by titles. Provide a list of project titles.')
+    parser.add_argument('--milestones-as-chapters', required=True, help='Treat milestones as chapters in the generated documentation.')
+    parser.add_argument('--repositories', required=True, help='JSON string defining the repositories to be included in the documentation generation.')
+
+    args = parser.parse_args()
+
+    env_vars = {
+        'GITHUB_TOKEN': args.github_token,
+        'PROJECT_STATE_MINING': args.project_state_mining,
+        'PROJECTS_TITLE_FILTER': args.projects_title_filter,
+        'MILESTONES_AS_CHAPTERS': args.milestones_as_chapters,
+        'REPOSITORIES': args.repositories
+    }
+
+    return env_vars
+
+
+def run_script(script_name, env_vars):
+    """ Helper function to run a Python script with environment variables using subprocess """
+    try:
+        # Setting up the environment for subprocess
+        env = os.environ.copy()
+        env.update(env_vars)
+
+        # Running the script with updated environment
+        result = subprocess.run(['python3', script_name], env=env, text=True, capture_output=True, check=True)
```

Comment:
calling Python from Python using subprocess is definitely not that common. Why not to call it directly? If env vars are the only reason, you can encapsulate them into a data class (I proposed something like that already in: https://github.com/AbsaOSS/living-doc-generator/issues/4#issuecomment-2115300899) and pass to each main function of each module, just an example.

Reply:
@MobiTikula Please address it in SPIKE for refactoring.

Reply:
See the solution proposal linked on following review comment.